### PR TITLE
#1553 Fix S4226: allow extensions on interfaces

### LINKED ISSUE #1553: Fix S4226: False positive for interfaces
### Description

> Extensions should be in separate namespaces.
> It makes little sense to create an extension method when it is possible to just add that method to the type itself.

However, adding a method is impossible when an interface is involved (at least for C# language versions prior to v8).

### Expected behavior

S4226 is not triggered for interfaces in language versions that don't support default method implementations.

### Known workarounds

N/A

### Related information

* SonarC# Version: 7.2 (build 5463)
* Visual Studio Version: 15.7.5
* If running through the Scanner for MSBuild, its version: 4.3.1.1372

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Fix #1553

### CODE REVIEW

**Comment by andrei-epure-sonarsource** [comment]
Must update ITs

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.cs [hunk 1]
```diff
@@ -48,4 +48,18 @@ static void Baz(this SomeNonExistingClass snec)
         {
         }
     }
-}
\ No newline at end of file
+}
+
+namespace OtherPackageNamespace
+{
+    interface Foo
```
Comment (Evangelink) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.cs]: [minor] It's a convention in C# to prefix interfaces by `I`.

File: sonaranalyzer-dotnet/its/expected/akka.net/Akka-{5DEDDF90-37F0-48D3-A0B0-A5CBD8A7E377}-S4226.json [hunk 2]
```diff
@@ -108,357 +43,6 @@
...
-"endLine":  30,
-"endColumn":  41
-}
-}
-},
-{
-"id":  "S4226",
-"message":  "Either move this extension to another namespace or move the method inside the type itself.",
-"location":  {
 "uri":  "sources\akka.net\src\core\Akka\Actor\SupervisorStrategy.cs",
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/its/expected/akka.net/Akka-{5DEDDF90-37F0-48D3-A0B0-A5CBD8A7E377}-S4226.json]: this is an enum and should not be reported, I don't understand why it is reported

Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/its/expected/akka.net/Akka-{5DEDDF90-37F0-48D3-A0B0-A5CBD8A7E377}-S4226.json]: I forgot to run update-expected :(

**Pull request merged**